index.js

```diff
@@ -10,125 +10,127 @@ import {
-  const selectedValue = event.target.id;
-  switch (selectedValue) {
-    case COMPLETE:
-      renderTasksByStatus(true);
-      break;
-    case INCOMPLETE:
-      renderTasksByStatus(false);
-      break;
-    default:
-      renderTodoList();
+import { TASK_PER_PAGE, START_INDEX, ALL, COMPLETE } from "./const.js";
+
+let searchText = "";
+let filterStatus;
+let searchResult;
+let currentPage = 1;
+let totalPageCount;
+
+export const disableFilterBar = () => {
+  if (todos.length === 0) {
+    tabs.forEach((tab) => {
+      tab.setAttribute("disabled", "");
+      tab.classList.remove("selected");
+    });
+
+    return;
   }
 
-  const tabs = document.querySelectorAll(".filter-tab");
-
-  tabs.forEach((tab) => tab.classList.remove("selected"));
+  tabs.forEach((tab) => tab.removeAttribute("disabled"));
+};
 
-  event.target.classList.add("selected");
+const paginationFilter = (todos) => {
+  totalPageCount = Math.ceil(todos.length / TASK_PER_PAGE);
+  renderLoadMoreButton(todos);
+  return todos.slice(START_INDEX, TASK_PER_PAGE * currentPage);
 };
 
-const renderTasksByStatus = (isDone) => {
-  keepCard1Elements();
-  filteredTodos = filteredTasks.length
-    ? filteredTasks.filter((todo) => todo.isDone === isDone)
-    : todos.filter((todo) => todo.isDone === isDone);
+const handleLoadMoreTask = () => {
+  if (todos && currentPage < totalPageCount) {
+    currentPage++;
+  }
 
-  filteredTodos.map((todo) => {
-    const newTask = createTaskElement(todo);
-    taskList.insertBefore(newTask, taskList.children[1]);
-  });
+  renderTodoList(todos);
 };
 
-export const keepCard1Elements = () => {
-  const elementsToKeep = [];
+export const renderLoadMoreButton = (todos) => {
+  if (todos.length >= TASK_PER_PAGE && currentPage !== totalPageCount) {
+    loadMoreButton.classList.remove("hide"),
+      loadMoreButton.classList.add("show");
 
-  for (let i = 0; i < taskList.children.length; i++) {
-    if (taskList.children[i].id === "card1") {
-      elementsToKeep.push(taskList.children[i]);
-    }
+    return;
   }
 
-  taskList.innerHTML = "";
+  loadMoreButton.classList.remove("show"), loadMoreButton.classList.add("hide");
 
-  elementsToKeep.map((element) => {
-    taskList.appendChild(element);
-  });
+  return;
 };
 
-const debounce = (handleSearch, delay) => {
-  let timer;
+const todoStateFilter = (status = ALL, todos) => {
+  if (status === ALL) {
+    return todos;
+  } else if (status === COMPLETE) {
+    return todos.filter((todo) => todo.isDone === true);
+  }
+  return todos.filter((todo) => todo.isDone === false);
+};
 
-  return function (...args) {
-    clearTimeout(timer);
+const searchFilter = (searchText, todos) => {
+  if (searchText !== "") {
+    return todos.filter((task) =>
+      task.value.toLowerCase().includes(searchText)
+    );
+  }
+  return todos;
+};
 
-    timer = setTimeout(() => {
-      handleSearch(...args);
-    }, delay);
+const debounce = (searchFilter, delay) => {
```

Comment:
`debounce` is a utility function. Let's move it to a new module. e.g: `utility.js` and export it from there.